ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -59,20 +63,23 @@
       .put(double.class, SchemaBuilder::float64)
       .build();
 
-  private static final Map<Pair<Schema.Type, Schema.Type>, Schema> ARITHMETIC_TYPE_MAPPINGS =
-      ImmutableMap.<Pair<Schema.Type, Schema.Type>, Schema>builder()
-          .put(new Pair<>(Schema.Type.INT64, Schema.Type.INT64), Schema.OPTIONAL_INT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.INT32, Schema.Type.INT64), Schema.OPTIONAL_INT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.INT64, Schema.Type.INT32), Schema.OPTIONAL_INT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.INT32, Schema.Type.INT32), Schema.OPTIONAL_INT32_SCHEMA)
-          .put(new Pair<>(Schema.Type.FLOAT64, Schema.Type.FLOAT64), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.FLOAT64, Schema.Type.INT32), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.INT32, Schema.Type.FLOAT64), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.FLOAT64, Schema.Type.INT64), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.INT64, Schema.Type.FLOAT64), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.FLOAT32, Schema.Type.FLOAT64), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.FLOAT64, Schema.Type.FLOAT32), Schema.OPTIONAL_FLOAT64_SCHEMA)
-          .put(new Pair<>(Schema.Type.STRING, Schema.Type.STRING), Schema.OPTIONAL_STRING_SCHEMA)
+  private static final Ordering<Schema.Type> ARITHMETIC_TYPE_ORDERING = Ordering.explicit(
+      ImmutableList.of(
+          Schema.Type.INT8,
+          Schema.Type.INT16,
+          Schema.Type.INT32,
+          Schema.Type.INT64,
+          Schema.Type.FLOAT32,
+          Schema.Type.FLOAT64
+      )
+  );
+
+  private static final NavigableMap<Schema.Type, Schema> TYPE_TO_SCHEMA =
+      ImmutableSortedMap.<Schema.Type, Schema>orderedBy(ARITHMETIC_TYPE_ORDERING)
+          .put(Schema.Type.INT32, Schema.OPTIONAL_INT32_SCHEMA)
+          .put(Schema.Type.INT64, Schema.OPTIONAL_INT64_SCHEMA)
+          .put(Schema.Type.FLOAT32, Schema.OPTIONAL_FLOAT64_SCHEMA)
```

Comment:
Should this be `Schema.OPTIONAL_FLOAT32_SCHEMA` instead...?

Reply:
if you examine the legacy code, we never resolved anything to `FLOAT32_SCHEMA`. Not sure if this was intentional or not, but I kept it that way (in fact, we never resolved anything that involved FLOAT32 and anything other than FLOAT64)

Reply:
Weird, but that makes sense. (Thanks for explaining!) Curious if anyone knows why this is the case though?